src/Repository/DonationRepository.php

```diff
@@ -26,4 +27,23 @@ public function findByEmailAddressOrderedByDonatedAt(string $email, string $orde
             ['donatedAt' => $order]
         );
     }
+
+    /**
+     * @return Donation[]
+     */
+    public function findAllSubscribedDonationByEmail(string $email): array
+    {
+        return $this->createQueryBuilder('donation')
+            ->andWhere('donation.emailAddress = :email')
+            ->andWhere('donation.duration != :duration')
+            ->andWhere('donation.subscriptionEndedAt IS NULL')
+            ->setParameters([
+                'email' => $email,
+                'duration' => PayboxPaymentSubscription::NONE,
+            ])
+            ->orderBy('donation.donatedAt', 'ASC')
+            ->getQuery()
+            ->getResult()
```

Comment:
why `getResult` if you can have juste only one subscription?

Reply:
it could have multiple subscription, actualy it's not blocked